tests/out/quad-vert.Vertex.glsl

```diff
@@ -40,7 +41,12 @@ void main() {
     a_uv = a_uv1;
     a_pos = a_pos1;
     main1();
-    type10 _tmp_return = type10(v_uv, perVertexStruct.gen_gl_Position, perVertexStruct.gen_gl_PointSize, perVertexStruct.gen_gl_ClipDistance, perVertexStruct.gen_gl_CullDistance);
+    vec2 _expr10 = v_uv;
+    vec4 _expr11 = perVertexStruct.gen_gl_Position;
+    float _expr12 = perVertexStruct.gen_gl_PointSize;
+    [1] _expr13 = perVertexStruct.gen_gl_ClipDistance;
```

Comment:
That's looks strange. We needed to fix load expression logic for glsl-out if we want to merge these changes.

Reply:
@Gordon-F would you be able to help with this! I don't want to leave #914 hanging in the air

Reply:
We could make this logic less aggressive by checking the pointer inside `Expression::Load { pointer }`. If it's pointing to `Expression::Constant` or `Expression::LocalVariable` we should return 1 bake count, otherwise 2 (default).

@paulgb Would you like to fix this in such a way? Also, it would be awesome to add your case to naga snapshot testing.

Reply:
I'm happy to add this, but `bake_ref_count` only has access to the `Handle` reference, not the arena needed to look it up.

I wonder if a cleaner approach would be to add a boolean field to `ExpressionInfo` called `may_vary`, set to  `true` iff the expression is a `Load` that depends on a global or local variable. The backend would then bake an expression if `may_vary || min_ref_count < expression_info.ref_count`.

Reply:
I made an attempt at the latter approach. It does fix the GLSL validation problem. It's a separate branch, but I could bring it into this PR if it's a desirable approach.

https://github.com/paulgb/naga/commit/3a11e22933b1c13efdc0ba92ec9563a9430e6551

Reply:
@kvark @paulgb Since we will have to change the public api, maybe it's easier to add `Arena<Expression>` reference to bake_ref_count ? It is used only in backends. 

Anyway, none of these solutions looks like a correct fix for this situation. We should return to this later (maybe with https://github.com/gfx-rs/naga/issues/392).

Reply:
Is this still a problem when rebased on top of master? ae367f6b74f34f3f7acbf7c8db0e03dcd37f74e4 could be affecting this

Reply:
It's still a problem on master. We can't handle arrays typed builtin (https://github.com/gfx-rs/naga/issues/851) in `glsl-out`.

Before:
```glsl
[1] _expr13 = perVertexStruct.gen_gl_ClipDistance;
```

After master rebase:
```
float _expr13 = perVertexStruct.gen_gl_ClipDistance;
```

Error:
```
Validating quad-vert.Vertex.glsl
stdin
ERROR: 0:43: '=' :  cannot convert from ' global 1-element array of highp float' to ' temp highp float'
ERROR: 0:43: '' : compilation terminated 
ERROR: 2 compilation errors.  No code generated.
```

Open a new PR #925 with another fix of this problem without validation errors.

Reply:
Can we just fix the GLSL variable declarations to support arrays better?
Would this work if it looked like `float _expr13[1] = perVertexStruct.gen_gl_ClipDistance;`?
In the worst case, we'd do what MSL backend does, and generate something like:
```
float _expr13[1] = { perVertexStruct.gen_gl_ClipDistance[0] };
```